lib/mpl_toolkits/mplot3d/tests/test_axes3d.py

```diff
@@ -2124,3 +2124,105 @@ def test_panecolor_rcparams():
                          'axes3d.zaxis.panecolor': 'b'}):
         fig = plt.figure(figsize=(1, 1))
         fig.add_subplot(projection='3d')
+
+
+def test_mutating_input_arrays_y_and_z():
+    fig = plt.figure()
+    ax = fig.add_subplot(111, projection='3d')
+    x = [1, 2, 3]
+    y = [0.0, 0.0, 0.0]
+    z = [0.0, 0.0, 0.0]
+    pl, = ax.plot(x, y, z, 'o-')
+
+    ax.set_ylim([0, 4])
+    ax.set_zlim([0, 4])
+
```

Comment:
For clarity one may consider do a `draw_without_rendering` here and check that the results are as expected (then mutate and check that they have not been modified).